stdlib/public/runtime/Demangle.cpp

```diff
@@ -634,3 +635,64 @@ swift::_swift_buildDemanglingForMetadata(const Metadata *type,
   // Not a type.
   return nullptr;
 }
+
+// NB: This function is not used directly in the Swift codebase, but is
+// exported for Xcode support and is used by the sanitizers. Please coordinate
+// before changing.
+//
+/// Demangles a Swift symbol name.
+///
+/// \param mangledName is the symbol name that needs to be demangled.
+/// \param mangledNameLength is the length of the string that should be
+/// demangled.
+/// \param outputBuffer is the user provided buffer where the demangled name
+/// will be placed. If nullptr, a new buffer will be malloced. In that case,
+/// the user of this API is responsible for freeing the returned buffer.
+/// \param outputBufferSize is the size of the output buffer. If the demangled
+/// name does not fit into the outputBuffer, the output will be truncated and
+/// the size will be updated, indicating how large the buffer should be.
+/// \param flags can be used to select the demangling style. TODO: We should
+//// define what these will be.
+/// \returns the demangled name. Returns nullptr if the input String is not a
+/// Swift mangled name.
+SWIFT_RUNTIME_EXPORT
```

Comment:
`SWIFT_RUNTIME_EXPORT` is for declarations. This definition should not use it.

Reply:
I gather that a declaration for this function wasn't desired because it's not used by the other Swift code. Should we stick a declaration in the header anyway? I don't think there's anything that says it *must* not be used if someone decides it would be useful.